Use aria-describedby in InputGroup

torkjels opened the pull request:
## Beskrivelse

Legger til `aria-describedby` i `InputGroup` når man definerer `fieldMessage` og/eller `description`. Skriver også om `InputGroup` til en funksjonell komponent.

## Motivasjon og kontekst

Gjør at feilmeldinger (og alle andre `fieldMessage`) og beskrivelser blir riktig koblet til feltene de beskriver slik at det blir enklere for skjermlesere å navigere seg gjennom skjemaer.

## Testing

Jeg gjorde endringer i `InputGroup.md`, kjørte opp `npm start` og testa der i Chrome. I tillegg skrev jeg noen tester og utvidet noen i `InputGroup.spec.js`.

MariaOverlierBerg commented on packages/ffe-form-react/src/InputGroup.spec.js:
```diff
@@ -174,4 +211,21 @@ describe('<InputGroup>', () => {
         );
     });
+
+    it('sets aria-describedby with two ids when both fieldMessage and description is defined', () => {
```
Veldig pirkete med burde det være "fieldMessage and description **are** defined" i stedet for is på linje 215 og 227? :smile:

MariaOverlierBerg reviewed (dismissed):
Veldig bra at dette kommer på plass. Veldig fine tester også! :clap:

ivarni commented on packages/ffe-form-react/src/InputGroup.js:
```diff
@@ -9,7 +9,15 @@ import ErrorFieldMessage from './ErrorFieldMessage';
+        this.fieldMessageId = props.fieldMessage
+            ? (props.fieldMessage.props && props.fieldMessage.props.id) ||
+              `${id}-fieldmessage`
+            : undefined;
```
Når du gjør dette i constructoren vil det vel bare finnes en `fieldMessageId` i de tilfeller der man sender inn en `fieldMessage` første gang man rendrer komponenten. Dette ser ut som det hadde passet bedre i en `useEffect` hook eller lignende, så kan komponenten skrives om til en functional component i samme slengen. 

Eventuelt lage en `fieldMessageId` selvom `props.fieldMessage` ikke er satt ved mount.

torkjels replied:
Veldig godt poeng, jeg fikser!

ivarni replied:
Muligens litt mer styr å lage test for, siden man må forholde seg til en lifecycle med endring av props. Jeg husker ikke lenger hvordan det er med enzyme, er en stund siden jeg har brukt det nå.

torkjels replied:
Da var det fikset, @ivarni.

ivarni replied:
Nice!

torkjels committed:
feat(ffe-form-react): add `aria-describedby` to `InputGroup`

Pass an `aria-describedby` to the children of `InputGroup` that points to `fieldMessage` and/or `description` when one or both are defined. Also rewrite from class component to functional component.
